Recipes/Recipes.API/Controllers/RecipesController.cs

```diff
@@ -84,6 +98,70 @@ public async Task<ActionResult> DeleteRecipe(int recipeId)
         return NoContent();
     }
 
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult> CreateRecipe(RecipeDTO request)
+    {
+        try
+        {
+            await InsertNewTags(request.Tags);
+
+            RecipeEntity recipeEntity = await AssignTagsToRecipe(request);
+
+            _recipesRepository.Insert(recipeEntity);
+            await _recipesRepository.SaveChangesAsync();
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return NoContent();
+    }
+
+    [HttpPut("{recipeId:int}")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult> UpdateRecipe(int recipeId, [FromBody] RecipeDTO request)
+    {
+        RecipeEntity? oldRecipe;
+        
+        try
+        {
+            oldRecipe = _recipesRepository.GetById(recipeId);
+
+            if(oldRecipe == null)
+            {
+                return StatusCode(StatusCodes.Status404NotFound);
+            }
+
+            await InsertNewTags(request.Tags);
+
+            RecipeEntity newRecipe = await AssignTagsToRecipe(request);
+
+            oldRecipe.Title = newRecipe.Title;
+            oldRecipe.AuthorId = newRecipe.AuthorId;
+            oldRecipe.Description = newRecipe.Description;
+            oldRecipe.CookingTime = newRecipe.CookingTime;
+            oldRecipe.Servings = newRecipe.Servings;
+            oldRecipe.Ingredients = newRecipe.Ingredients;
+            oldRecipe.Instructions = newRecipe.Instructions;
+            oldRecipe.Image = newRecipe.Image;
+            oldRecipe.Tags = newRecipe.Tags;
```

Comment:
I did some refactoring but could not really escape from this. The problem is that I need the `oldRecipe`'s pointer so that I can pass it later into the `Update` in `RecipesRepository`.